### PR TITLE
New Recipe: libCEED v0.7.0

### PR DESCRIPTION
**Pull request by pazner**

### CODE REVIEW

File: L/libCEED/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Linux(:x86_64, libc=:glibc),
+    Linux(:aarch64, libc=:glibc),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:powerpc64le, libc=:glibc),
+    Linux(:i686, libc=:musl),
+    Linux(:x86_64, libc=:musl),
+    Linux(:aarch64, libc=:musl),
+    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
+    MacOS(:x86_64)
+]
```
Comment (Gnimuc) [inline L/libCEED/build_tarballs.jl:36]:
```suggestion
platforms = supported_platforms(exclude= x -> typeof(x) == Windows || typeof(x) == FreeBSD)
```
or use the new platform ctors, e.g. `Platform("x86_64", "linux"; libc="glibc")`.

Comment (pazner) [reply L/libCEED/build_tarballs.jl]: Thanks. I was a bit confused by the error message ```LoadError: ArgumentError: Platform `i686-linux-glibc` is not an officially supported platform```.

Comment (pazner) [reply L/libCEED/build_tarballs.jl]: It looks like the CI is still trying to build for `i686-w64-mingw32` and `x86_64-w64-mingw32`. Is there something I can do to fix this?

File: L/libCEED/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+cd libCEED
+sed -i "s/MEMCHK :=.*/MEMCHK := 0/g" Makefile 
+sed -i "s/CC_VENDOR :=.*/CC_VENDOR := gcc/g" Makefile 
+make -j$ncore
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(exclude=x->typeof(x)==Windows)
```
Comment (Gnimuc) [inline L/libCEED/build_tarballs.jl:25]:
```suggestion
platforms = supported_platforms(exclude=[Platform("x86_64", "windows"),Platform("i686", "windows")])
```

File: L/libCEED/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+cd libCEED
+sed -i "s/MEMCHK :=.*/MEMCHK := 0/g" Makefile 
+sed -i "s/CC_VENDOR :=.*/CC_VENDOR := gcc/g" Makefile 
+make -j$ncore
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(exclude=[Platform("x86_64", "windows"),Platform("i686", "windows")])
```
Comment (giordano) [inline L/libCEED/build_tarballs.jl:25]: This is simpler and more future-proof
```suggestion
platforms = supported_platforms(exclude=Sys.iswindows)
```

File: L/libCEED/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    GitSource("https://github.com/CEED/libCEED.git", "6134893e0067dc2a87bbbe22a551ae134ad5b885")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd libCEED
+sed -i "s/MEMCHK :=.*/MEMCHK := 0/g" Makefile 
+sed -i "s/CC_VENDOR :=.*/CC_VENDOR := gcc/g" Makefile 
+make -j$ncore
```
Comment (giordano) [inline L/libCEED/build_tarballs.jl:19]: This is simpler and doesn't require editing any file:
```suggestion
cd $WORKSPACE/srcdir/libCEED
make -j${nproc} MEMCHK=0 CC_VENDOR=cc
```
Also, the number of process is `${nproc}` :wink:

File: L/libCEED/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+products = [
+    LibraryProduct("libceed", :libceed)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```
Comment (giordano) [inline L/libCEED/build_tarballs.jl:37]: Any particular reason for using GCC 9?

File: L/libCEED/build_tarballs.jl (hunk 4 above)
Comment (pazner) [reply L/libCEED/build_tarballs.jl]: Good simplifications, thanks.

File: L/libCEED/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/CEED/libCEED.git", "6134893e0067dc2a87bbbe22a551ae134ad5b885")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/libCEED
+make -j${nproc} MEMCHK=0 CC_VENDOR=gcc
+make install
```
Comment (giordano) [inline L/libCEED/build_tarballs.jl:17]: Same argument also to `make install`:
```suggestion
make -j${nproc} MEMCHK=0 CC_VENDOR=cc
make install MEMCHK=0 CC_VENDOR=cc
```
Also, don't force GCC as compiler, when building for FreeBSD and macOS we don't want to use that, if possible

Comment (pazner) [reply L/libCEED/build_tarballs.jl]: The variable `CC_VENDOR` determines the compiler flags, but not the compiler itself. `CC_VENDOR=cc` will not have an effect. The flags for gcc and clang are the same, so I think this should be fine. There is some logic in the Makefile to set `CC_VENDOR` automatically based on the output of `$(CC) --version`, but it was not working properly.

Comment (giordano) [reply L/libCEED/build_tarballs.jl]: Ok, looking at the log it's actually building with `cc`, which is what we want.  All good then, thanks!

**Pull request merged**